### PR TITLE
Allow to configure TTL, interval and timeout

### PR DESCRIPTION
**Pull request by ldez**
Allow to configure TTL, interval or timeout (if they are available) with env var or with a DNSProvider constructor:
- `XXX_TTL`
- `XXX_PROPAGATION_TIMEOUT`
- `XXX_POLLING_INTERVAL`
- `XXX_HTTP_TIMEOUT`

Related to #474, #475

### CODE REVIEW

File: providers/dns/alidns/alidns.go [hunk 1]
```diff
@@ -25,48 +51,74 @@ type DNSProvider struct {
...
 	}
 
-	client, err := alidns.NewClientWithAccessKey(regionID, apiKey, secretKey)
+	conf := sdk.NewConfig().WithTimeout(config.HTTPTimeout)
+	credential := credentials.NewAccessKeyCredential(config.APIKey, config.SecretKey)
+
+	client, err := alidns.NewClientWithOptions(config.RegionID, conf, credential)
 	if err != nil {
-		return nil, fmt.Errorf("AliDNS: credentials failed: %v", err)
+		return nil, fmt.Errorf("alicloud: credentials failed: %v", err)
```
Comment (wyattjoh) [inline providers/dns/alidns/alidns.go]: Another vote for replacing these with `github.com/pkg/errors.Wrap`?

File: providers/dns/azure/azure.go [hunk 2]
```diff
@@ -91,59 +121,55 @@ func (d *DNSProvider) Present(domain, token, keyAuth string) error {
...
-// Returns the relative record to the domain
-func toRelativeRecord(domain, zone string) string {
-	return acme.UnFqdn(strings.TrimSuffix(domain, zone))
+	_, err = rsc.CreateOrUpdate(ctx, d.config.ResourceGroup, zone, relative, dns.TXT, rec, "", "")
+	return fmt.Errorf("azure: %v", err)
 }
 
 // CleanUp removes the TXT record matching the specified parameters
 func (d *DNSProvider) CleanUp(domain, token, keyAuth string) error {
+	ctx := context.Background()
```
Comment (wyattjoh) [inline providers/dns/azure/azure.go]: It should accept a context rather than using background.

File: providers/dns/dnsimple/dnsimple_test.go [hunk 3]
```diff
@@ -44,6 +46,8 @@ func TestNewDNSProviderValid(t *testing.T) {
 	defer restoreEnv()
 	os.Setenv("DNSIMPLE_OAUTH_TOKEN", "123")
 
+	acme.UserAgent = "lego"
```
Comment (wyattjoh) [inline providers/dns/dnsimple/dnsimple_test.go]: Why are we settings this in a test?

File: providers/dns/azure/azure.go (hunk 2 above)
Comment (ldez) [reply providers/dns/azure/azure.go]: it's not possible because this method is a part of an interface

File: providers/dns/alidns/alidns.go (hunk 1 above)
Comment (ldez) [reply providers/dns/alidns/alidns.go]: nope, I don't think we need to use `github.com/pkg/error`

File: providers/dns/dnsimple/dnsimple_test.go (hunk 3 above)
Comment (ldez) [reply providers/dns/dnsimple/dnsimple_test.go]: because it's the goal of this test: 
```go
assert.Equal(t, "lego", provider.client.UserAgent)
```

Comment (wyattjoh) [reply providers/dns/dnsimple/dnsimple_test.go]: Missed that, sorry!

**Pull request merged**